### PR TITLE
Go 1.19 support and linter fixes

### PR DESCRIPTION
**Pull request by berndverst**
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

This switches contrib to use Go 1.19.
That means Golangci lint v1.48.0 is now required.

```bash
 curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.48.0
```

I painstakingly took the time to address **every** linter failure the v1.48.0 linter revealed.

The workflows have been updated in this PR.

As part of this PR I had to upgrade the IPFS bindings dependencies to compile the project. All tests still pass. (verified with `IPFS_TEST=y go test -v --count=1 ./bindings/ipfs`)

### CODE REVIEW

**Comment by berndverst** [comment]
image

**Comment by berndverst** [comment]
image

**Comment by berndverst** [comment]
Since this updates the IPFS tests which can only be run by setting a flag, here is the result of that:

```
IPFS_TEST=y go test -v --count=1 ./bindings/ipfs
=== RUN   TestSingleNodeGlobalNetwork
=== RUN   TestSingleNodeGlobalNetwork/init_node
time="2022-08-17T20:48:11.47576-07:00" level=info msg="Initialized a new IPFS repo at path /var/folders/x9/9dhxnxcd5bjbyktd4h_vhgyc0000gn/T/TestSingleNodeGlobalNetwork2631730773/001" instance=Bernds-MacBook-Pro.local scope=tests type=log ver=unknown
time="2022-08-17T20:48:11.609045-07:00" level=info msg="Opened IPFS repo at path /var/folders/x9/9dhxnxcd5bjbyktd4h_vhgyc0000gn/T/TestSingleNodeGlobalNetwork2631730773/001" instance=Bernds-MacBook-Pro.local scope=tests type=log ver=unknown
time="2022-08-17T20:48:12.122958-07:00" level=info msg="Started IPFS node 12D3KooWJkDeF86ZpB8R6n6HCU9wE9PXF4Hgy5Naqi2sf7fuitaH" instance=Bernds-MacBook-Pro.local scope=tests type=log ver=unknown
=== RUN   TestSingleNodeGlobalNetwork/get_operation
=== RUN   TestSingleNodeGlobalNetwork/get_operation/empty_path
=== RUN   TestSingleNodeGlobalNetwork/get_operation/retrieve_document_by_CID
=== RUN   TestSingleNodeGlobalNetwork/get_operation/retrieve_document_by_IPLD
=== RUN   TestSingleNodeGlobalNetwork/get_operation/cannot_retrieve_folder
=== RUN   TestSingleNodeGlobalNetwork/get_operation/retrieve_files
=== RUN   TestSingleNodeGlobalNetwork/ls_operation
=== RUN   TestSingleNodeGlobalNetwork/ls_operation/empty_path
=== RUN   TestSingleNodeGlobalNetwork/ls_operation/list_by_CID
=== RUN   TestSingleNodeGlobalNetwork/ls_operation/list_by_IPLD
=== RUN   TestSingleNodeGlobalNetwork/pin_operations
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_nothing_is_pinned_on_a_new_node
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-add:_pin_file_by_CID
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_list_added_pin
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-add:_pin_file_by_IPLD
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-add:_recursively_pin_folder
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_list_added_folder
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-rm:_remove_file
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_updated_after_removed_file
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-rm:_recursively_remove_folder
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-rm:_remove_explicitly-pinned_file
=== RUN   TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_updated_list_is_empty
=== RUN   TestSingleNodeGlobalNetwork/create_operation
=== RUN   TestSingleNodeGlobalNetwork/create_operation/add_with_default_options
=== RUN   TestSingleNodeGlobalNetwork/create_operation/add_with_CID_v1
=== RUN   TestSingleNodeGlobalNetwork/create_operation/added_files_are_pinned
=== RUN   TestSingleNodeGlobalNetwork/create_operation/add_without_pinning
=== RUN   TestSingleNodeGlobalNetwork/create_operation/pinned_documents_haven't_changed
=== RUN   TestSingleNodeGlobalNetwork/create_operation/add_inline
--- PASS: TestSingleNodeGlobalNetwork (3.10s)
    --- PASS: TestSingleNodeGlobalNetwork/init_node (0.65s)
    --- PASS: TestSingleNodeGlobalNetwork/get_operation (0.92s)
        --- PASS: TestSingleNodeGlobalNetwork/get_operation/empty_path (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/get_operation/retrieve_document_by_CID (0.13s)
        --- PASS: TestSingleNodeGlobalNetwork/get_operation/retrieve_document_by_IPLD (0.09s)
        --- PASS: TestSingleNodeGlobalNetwork/get_operation/cannot_retrieve_folder (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/get_operation/retrieve_files (0.70s)
    --- PASS: TestSingleNodeGlobalNetwork/ls_operation (0.04s)
        --- PASS: TestSingleNodeGlobalNetwork/ls_operation/empty_path (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/ls_operation/list_by_CID (0.04s)
        --- PASS: TestSingleNodeGlobalNetwork/ls_operation/list_by_IPLD (0.00s)
    --- PASS: TestSingleNodeGlobalNetwork/pin_operations (0.58s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_nothing_is_pinned_on_a_new_node (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-add:_pin_file_by_CID (0.08s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_list_added_pin (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-add:_pin_file_by_IPLD (0.11s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-add:_recursively_pin_folder (0.12s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_list_added_folder (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-rm:_remove_file (0.10s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_updated_after_removed_file (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-rm:_recursively_remove_folder (0.08s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-rm:_remove_explicitly-pinned_file (0.08s)
        --- PASS: TestSingleNodeGlobalNetwork/pin_operations/pin-ls:_updated_list_is_empty (0.00s)
    --- PASS: TestSingleNodeGlobalNetwork/create_operation (0.77s)
        --- PASS: TestSingleNodeGlobalNetwork/create_operation/add_with_default_options (0.20s)
        --- PASS: TestSingleNodeGlobalNetwork/create_operation/add_with_CID_v1 (0.24s)
        --- PASS: TestSingleNodeGlobalNetwork/create_operation/added_files_are_pinned (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/create_operation/add_without_pinning (0.14s)
        --- PASS: TestSingleNodeGlobalNetwork/create_operation/pinned_documents_haven't_changed (0.00s)
        --- PASS: TestSingleNodeGlobalNetwork/create_operation/add_inline (0.19s)
PASS
ok      github.com/dapr/components-contrib/bindings/ipfs        3.132s
```

File: secretstores/local/file/filestore_test.go [hunk 1]
```diff
@@ -3,7 +3,9 @@ Copyright 2021 The Dapr Authors
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
-    http://www.apache.org/licenses/LICENSE-2.0
+
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (berndverst) [inline secretstores/local/file/filestore_test.go:7]: For some reason this is what the linter wants on several files. And the VS Code Go extension made this modification automatically via `gofumpt`.

Comment (ItalyPaleAle) [reply secretstores/local/file/filestore_test.go:7]: I know why this is happening. Go 1.19 is adding markdown-ish support to comments. It should be ok

File: .golangci.yml [hunk 2]
```diff
@@ -244,6 +244,7 @@ linters:
     - nestif
     - nlreturn
     - exhaustive
+    - exhaustruct
```
Comment (ItalyPaleAle) [inline .golangci.yml:247]: Thank you 😅

File: bindings/zeebe/command/publish_message_test.go [hunk 3]
```diff
@@ -76,8 +76,8 @@ func (cmd2 *mockPublishMessageCommandStep2) CorrelationKey(correlationKey string
 }
 
 // MessageId comes from the Zeebe client API and cannot be written as MessageID
-// Note that when the `stylecheck` linter is working again, this method will need "nolink:stylecheck" (can't change name to ID or it won't satisfy an interface)
-func (cmd3 *mockPublishMessageCommandStep3) MessageId(messageID string) commands.PublishMessageCommandStep3 {
+// Note that when the `stylecheck` linter is working again, this method will need "nolint:stylecheck" (can't change name to ID or it won't satisfy an interface)
```
Comment (ItalyPaleAle) [inline bindings/zeebe/command/publish_message_test.go:79]: This line can be removed. Or changed to:

```
This must be named `MessageId` (with apologies to the linter 💄 #SorryNotSorry) or it won't satisfy the interface
```

File: state/mysql/mysql.go [hunk 4]
```diff
@@ -245,6 +245,11 @@ func (m *MySQL) ensureStateTable(stateTableName string) error {
 		// never need to pass it in.
 		// eTag is a UUID stored as a 36 characters string. It needs to be passed
 		// in on inserts and updates and is used for Optimistic Concurrency
+
+		// adding a basic precaution to ensure our SQL query is not hijacked
+		stateTableName = strings.Split(strings.Split(stateTableName, ";")[0], " ")[0]
+
+		//nolint:gosec
 		createTable := fmt.Sprintf(`CREATE TABLE %s (
```
Comment (ItalyPaleAle) [inline state/mysql/mysql.go:253]: Please change to use prepared statements:

```go
_, err = m.db.Exec(
	`CREATE TABLE $1 (
	id VARCHAR(255) NOT NULL PRIMARY KEY,
	value JSON NOT NULL,
	isbinary BOOLEAN NOT NULL,
	insertDate TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
	updateDate TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
	eTag VARCHAR(36) NOT NULL
	);`,
	stateTableName,
)
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Please change the MySQL thing, otherwise LGTM

File: bindings/alicloud/tablestore/tablestore.go [hunk 5]
```diff
@@ -262,11 +262,11 @@ func (s *AliCloudTableStore) create(req *bindings.InvokeRequest, resp *bindings.
 		TableName:     s.getTableName(req.Metadata),
 		PrimaryKey:    &tablestore.PrimaryKey{PrimaryKeys: pks},
 		Columns:       columns,
-		ReturnType:    tablestore.ReturnType_RT_NONE,
+		ReturnType:    tablestore.ReturnType_RT_NONE, //nolint:nosnakecase
```
[the remark this answers is not in the document]
Comment (berndverst) [reply bindings/alicloud/tablestore/tablestore.go:265]: For the entire file, yes it's something I was considering, but in general I don't want folks to use snakeCase :)

File: bindings/output_binding.go [hunk 6]
```diff
@@ -32,6 +32,6 @@ func PingOutBinding(outputBinding OutputBinding) error {
 	if outputBindingWithPing, ok := outputBinding.(health.Pinger); ok {
 		return outputBindingWithPing.Ping()
 	} else {
-		return fmt.Errorf("Ping is not implemented by this output binding")
+		return fmt.Errorf("Ping is not implemented by this output binding") //nolint:stylecheck
```
[the remark this answers is not in the document]
Comment (berndverst) [reply bindings/output_binding.go:35]: Probably, but I wasn't sure whether someone is relying on this, or whether the method name is case sensitive "Ping" (which could be different from "ping"), so I am keeping it for now. Will consider updating subsequently.

File: bindings/zeebe/command/publish_message_test.go (hunk 3 above)
Comment (berndverst) [reply bindings/zeebe/command/publish_message_test.go:79]:
```suggestion
```

File: bindings/output_binding.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (berndverst) [reply bindings/output_binding.go:35]:
```suggestion
		return fmt.Errorf("ping is not implemented by this output binding")
```

File: .golangci.yml (hunk 2 above)
Comment (berndverst) [reply .golangci.yml:247]: 🎉

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

File: state/mysql/mysql.go [hunk 7]
```diff
@@ -202,9 +202,7 @@ func (m *MySQL) ensureStateSchema() error {
 	if !exists {
 		m.logger.Infof("Creating MySql schema '%s'", m.schemaName)
 
-		createTable := fmt.Sprintf("CREATE DATABASE %s;", m.schemaName)
-
-		_, err = m.db.Exec(createTable)
+		_, err = m.db.Exec(`CREATE DATABASE ?`, m.schemaName)
```
Comment (ItalyPaleAle) [inline state/mysql/mysql.go:205]: TODO: @ItalyPaleAle  to fix

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by 1046102779** [comment]
@berndverst 

image

image


Could you please tell me if there were done manually, or  fix it with tools? I want to understand and learn, this is very useful

**Pull request merged**